src/routes/secret.go

```diff
@@ -0,0 +1,141 @@
+package routes
+
+import (
+	"net/http"
+
+	"github.com/dana-team/platform-backend/src/controllers"
+	"github.com/dana-team/platform-backend/src/types"
+	"github.com/gin-gonic/gin"
+	"go.uber.org/zap"
+	"k8s.io/client-go/kubernetes"
+)
+
+func CreateSecret(client kubernetes.Interface, logger *zap.Logger) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		namespace := c.Param("namespace")
+		if namespace == "" {
+			c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": "Namespace path parameter is required"})
+			return
+		}
+
+		var request types.CreateSecretRequest
+		if err := c.ShouldBindJSON(&request); err != nil {
+			c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+			return
+		}
+
+		context := c.Request.Context()
+		secretController := controllers.NewSecretController(client, context, logger)
+		response, err := secretController.CreateSecret(namespace, request)
+		if err != nil {
+			c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{"error": "Failed to create a new secret", "details": err.Error()})
```

Comment:
The status code  will be always internal server error the controller.  the Kubernetes client returns status code return the status code from the error

Reply:
Yep, thanks!